src/gui/raster/qgssinglebandpseudocolorrendererwidget.cpp

```diff
@@ -157,30 +192,58 @@ QgsRasterRenderer* QgsSingleBandPseudoColorRendererWidget::renderer()
   return renderer;
 }
 
-void QgsSingleBandPseudoColorRendererWidget::on_mAddEntryButton_clicked()
-{
-  QTreeWidgetItem* newItem = new QTreeWidgetItem( mColormapTreeWidget );
-  newItem->setText( 0, "0.0" );
-  newItem->setBackground( 1, QBrush( QColor( Qt::magenta ) ) );
-  newItem->setText( 2, tr( "Custom color map entry" ) );
-}
-
-void QgsSingleBandPseudoColorRendererWidget::on_mDeleteEntryButton_clicked()
+/** Generate labels from the values in the color map.
+ *  Skip labels which were manually edited (black text).
+ *  Text of generated labels is made gray
+ */
+void QgsSingleBandPseudoColorRendererWidget::autoLabel()
 {
-  QTreeWidgetItem* currentItem = mColormapTreeWidget->currentItem();
-  if ( currentItem )
+  bool discrete = mColorInterpolationComboBox->currentText() == tr( "Discrete" );
+  QString unit = mUnitLineEdit->text();
+  QString label = "";
+  int myTopLevelItemCount = mColormapTreeWidget->topLevelItemCount();
+  QTreeWidgetItem* myCurrentItem;
+  for ( int i = 0; i < myTopLevelItemCount; ++i )
   {
-    delete currentItem;
+    myCurrentItem = mColormapTreeWidget->topLevelItem( i );
+    //If the item is null or does not have a pixel values set, skip
+    if ( !myCurrentItem || myCurrentItem->text( 0 ) == "" )
```

Comment:
Instead of `myCurrentItem->text( 0 ) == ""` use `myCurrentItem->text( 0 ).isEmpty()` (much more efficient)